### DISCUSSION TITLE
How to build front and backend for production

### DISCUSSION THREAD

**Discussion by RitchieP** [alex8088/electron-vite#197] [Q&A] [answered]
This is my first time building an app that is about to go into real production. 

My application has the backend built right into the `src` folder. During development, I will run the frontend and backend manually in two different terminals.

Now that I want to package the whole application to be a production build, how should I configure `electron-build` or any configuration files needed so that when the user installs and runs the app, the backend and frontend will both run?

**Comment by alex8088** [comment, marked as answer]
This doesn't seem to work. You may need something like nestjs+electron. But I think you need to understand electron's process model first. You can run your service in the main process of electron, but this service must be based on nodejs.

**Reply by RitchieP** [reply]
Yes, my backend is based on NodeJS. Meaning that I will need to spawn my service in main process

**Reply by alex8088** [reply]
Yes, it can also be run directly in the electron main process.
